### PR TITLE
Gateway: More wallet support

### PR DESCRIPTION
**Pull request by gigamesh**
- Refactors wallet logic & global state to be more amenable to additional wallets
- Adds wallet choice modal
- Adds Coinbase Wallet (currently commented out until it can be tested & debugged)
- Removes $ gas estimates when not on mainnet (suggestion from Kevin to avoid confusion)

https://deploy-preview-59--optimism-gateway.netlify.app/

### CODE REVIEW

**Comment by snario** [comment]
When I open this in my browser I see this:

image

When I click Coinbase Wallet it opens up the App Store

**Comment by snario** [comment]
On Optimistic Kovan on Coinbase Wallet if I click "MetaMask" the button is colored but no UI action happens

File: packages/gateway.optimism.io/src/components/HeaderNav.tsx [hunk 1]
```diff
@@ -85,12 +66,13 @@ function HeaderNav({ isMobileDrawerOpen, openMobileDrawer, closeMobileDrawer }:
             _active={{
               bg: 'brandPrimaryAlphaPt1',
             }}
+            fontSize="inherit"
             cursor={'pointer'}
-            onClick={connectedNetwork ? () => openModal(modalTypes.DISCONNECT) : () => connectToLayer()}
+            onClick={() => openModal(connectedNetwork ? modalTypes.DISCONNECT : modalTypes.CHOOSE_WALLET)}
```
Comment (platocrat) [inline packages/gateway.optimism.io/src/components/HeaderNav.tsx:71]: sexyy

File: packages/gateway.optimism.io/src/components/Modal/ChooseWalletModal.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Text, Box, Heading, Image } from '@chakra-ui/react';
+import { SUPPORTED_WALLETS } from '../../web3/connectors';
+import { useWeb3React } from '@web3-react/core';
+
+function ChooseWalletModal() {
+  const { activate } = useWeb3React();
```
Comment (platocrat) [inline packages/gateway.optimism.io/src/components/Modal/ChooseWalletModal.tsx:7]: ❤️

File: packages/gateway.optimism.io/src/context/AppContext.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  allowanceAmount?: ethers.BigNumber;
+  handleApprove: () => void;
+  approvalHash: string | null;
+  depositHash: string | null;
+  withdrawalHash: string | null;
+  openModal: (type: string) => void;
+  closeModal: () => void;
+  isModalOpen: boolean;
+  currentModal?: string;
+};
```
Comment (platocrat) [inline packages/gateway.optimism.io/src/context/AppContext.tsx:45]: this is quite long. I'm sure you already know, but if we keep moving at this speed, this props type can start to get out of hand

**Review by platocrat — APPROVED** [review]
lgtm

File: packages/gateway.optimism.io/src/context/AppContext.tsx (hunk 3 above)
Comment (gigamesh) [reply packages/gateway.optimism.io/src/context/AppContext.tsx:45]: yea I can still clean this up a bit more

**Pull request merged**